### PR TITLE
Documentation, bug fixes, and minor features

### PR DESCRIPTION
**Pull request by mpinkert**
This pull request has a lot of added redone/documentation.

We also fix many bugs.

There's also a few new features, such as the `ij.py.get_window_manager` function or the ability to enter Maven endpoints as a list, rather than as a single string.

### CODE REVIEW

File: imagej/imagej.py [hunk 1]
```diff
@@ -630,6 +630,18 @@ def _format_value(self, value):
...
 
+        def get_window_manager(self):
+            """
+            Get the ImageJ1 window manager if legacy mode is enabled.  It may not work properly if in headless mode.
+            :return: WindowManager
+            """
+            if not ij.legacy_enabled:
+                raise ImportError("Your ImageJ installation does not support IJ1.  This function does not work.")
+            elif ij.ui().isheadless():
+                logging.warning("Headless mode not enabled.  The WindowManager may not function as expected.")
```
Comment (imagejan) [inline imagej/imagej.py]: Shouldn't this read "Headless mode enabled."?

File: imagej/imagej.py [hunk 2]
```diff
@@ -616,6 +631,18 @@ def _format_value(self, value):
...
+            else:
+                return WindowManager
+
```
Comment (ctrueden) [inline imagej/imagej.py]: In the following `window_to_xarray` method, let's make behavior use pure IJ2 if the legacy layer is not active.

File: imagej/imagej.py (hunk 1 above)
Comment (mpinkert) [reply imagej/imagej.py]: You are correct.

File: doc/Troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Common Errors
+
+## I ran a plugin and see an updated image, but the numpy array and dataset are unchanged.
+This bug occurs when using ImageJ1 plugins which update a corresponding ImagePlus.
+It can be worked around by calling
+```python
+import imagej
+ij = imagej.init('net.imagej:imagej+net.imagej:imagej-legacy')
+imp = ij.py.get_img_plus()
+# For a single modification, and after subsequent modifications calling
```
[the remark this answers is not in the document]
Comment (mpinkert) [reply doc/Troubleshooting.md]: I've changed this to 
```python
import imagej
ij = imagej.init(['net.imagej:imagej', 'net.imagej:imagej-legacy'])
imp = ij.py.get_img_plus()
```
Once you have `imp`, the `ImagePlus`, you can call subsequently use the `synchronize_ij1_to_ij2` function to update it.
```python
ij.py.synchronize_ij1_to_ij2(imp)
```

File: doc/Troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ij.py.synchronize_ij1_to_ij2(imp)
+```
+
+## The GUI does not work on mac
+There are no current solutions.  See [this thread](issues/23) for updates.
```
[the remark this answers is not in the document]
Comment (mpinkert) [reply doc/Troubleshooting.md]: Fix:
The default UI does not currently work on mac.  You can set up a Swing UI based GUI instead by initializing
with:

```python
import imagej
ij = imagej.init('/Applications/Fiji.app', headless=False)
ij.ui().showUI("swing")
```

See [this thread](issues/23) for additional information and updates.

File: doc/Usage.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+```python
+import imagej
+ij = imagej.init(['net.imagej:imagej', 'net.imagej.:imagej-legacy'])
+language_extension = 'ijm'
+result_script = ij.py.run_script(language_extension, macro, args)
+print(result_script.getOutput('greeting'))
+```
+
+Finally, running plugins works in the same manner as macros. You simply enter the plugin name as a string
```
[the remark this answers is not in the document]
Comment (mpinkert) [reply doc/Usage.md]: IdeaVim apparently doesn't like that command, for future reference.  I've fixed it up manually.

File: doc/Usage.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+Here is an example of using the "Mean" plugin to blur an image
+  
+```python
+import imagej
+ij = imagej.init(['net.imagej:Imagej', 'net.imagej:imagej-legacy'], headless=False)
+ij.py.run_macro("""run("Blobs (25K)");""")
+blobs = ij.py.get_image_plus()
+ij.py.show(blobs)
+
+plugin = 'Mean'
```
[the remark this answers is not in the document]
Comment (mpinkert) [reply doc/Usage.md]: I've changed it to use `sc.fiji:fiji:2.0.0-pre-10'

File: doc/Usage.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+import imagej
+import numpy as np
+ij = imagej.init()
+
+# numpy image
+img1 = np.zeros([10, 10])
+print(ij.py.dims(img1)) # (10, 10)
+
+# imagej image
+img2 = ij.py.to_java(img1)
```
[the remark this answers is not in the document]
Comment (mpinkert) [reply doc/Usage.md]: Converted, including in tutorial.

File: imagej/imagej.py (hunk 2 above)
Comment (mpinkert) [reply imagej/imagej.py]: Change:
```python
            if ij.legacy().isActive():
                imp = self.image_plus(sync=sync)
                return self._ij.py.from_java(imp)
            else:
                dataset = self._ij.dataset().getActiveDataset()
                return self._ij.py_from_java(dataset)
```

File: doc/Usage.md [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+In order to use ImageJ1 windows, you must also intialize PyImageJ with `headless=False`.  To work with windows, you can
+either use the convenience functions available in `ij.py`, use the 
+[ImageJ2 Window Service](https://javadoc.scijava.org/ImageJ/net/imagej/display/WindowService.html) through `ij.window()`
+or directly get the ImageJ1 `ij.WindowManager` using the `ij.py.get_window_manager()` function.  See the 
+[WindowManager Javadoc](https://javadoc.scijava.org/ImageJ1/index.html?ij/WindowManager.html) for the full function
+list.
+
+#### Convenience functions
+Current convenience functions include `get_image_plus` to get the `ij.ImagePlus` from the current window and
+`window_to_xarray` to convert the current window into an `xarray.DataArray`.
```
[the remark this answers is not in the document]
Comment (mpinkert) [reply doc/Usage.md]: Implemented all three.

File: imagej/imagej.py [hunk 9]
```diff
@@ -616,6 +631,18 @@ def _format_value(self, value):
             final_value = '[' + temp_value + ']'
             return final_value
 
+        def get_window_manager(self):
```
[the remark this answers is not in the document]
Comment (mpinkert) [reply imagej/imagej.py]: Renamed `get_window_manager` to `window_manager` and `get_image_plus` to `active_image_plus`, as above

File: imagej/imagej.py [hunk 10]
```diff
@@ -414,7 +430,7 @@ def _get_axis_num(self, xarr, axis):
             if numpy.isfortran(xarr.values):
                 return py_axnum
 
-            if xarr.dims[len(xarr.dims)-1] == 'c':
+            if xarr.dims[len(xarr.dims)-1].lower() in ['c', 'channel']:
```
[the remark this answers is not in the document]
Comment (mpinkert) [reply imagej/imagej.py]: Created a new helper functino
```python
        def _ends_with_channel_axis(self, xarr):
            ends_with_axis = xarr.dims[len(xarr.dims)-1].lower() in ['c', 'channel']
            return ends_with_axis
```

**Comment by ctrueden** [comment]
Thanks for all your work @mpinkert! Sorry this slipped through the cracks till now.

**Pull request merged**